testing/test_cache.py

```diff
@@ -87,7 +87,30 @@ def test_cachefuncarg(cache):
         assert result.ret == 0
         result.stdout.fnmatch_lines(["*1 passed*"])
 
-
+    def test_custom_cache_dirname(self, testdir):
+        testdir.makeini("""
+            [pytest]
+            cache_dirname = custom_cache_dirname
+        """)
+        testdir.makepyfile(test_errored='def test_error():\n    assert False')
+        testdir.runpytest()
+        assert os.path.exists('custom_cache_dirname')
```

Comment:
I'm surprised this is passing... does `testdir` change the cwd of the test?

You might be more explicit here either way:

```python
assert testdir.tmpdir.join('custom_cache_dirname').isdir()
```